runtime/src/main/java/org/corfudb/runtime/object/transactions/ConflictSetInfo.java

```diff
@@ -21,29 +26,32 @@
 
 
     /** Set of objects this conflict set conflicts with. */
-    protected Map<UUID, Set<Object>> conflicts = new HashMap<>();
+    protected Map<ICorfuSMRProxyInternal, Set<Object>> conflicts = new HashMap<>();
 
-    /** Get the conflict set for a specific stream.
-     * @param streamId      The stream to obtain hashed conflicts for.
-     * @return              A set of hashed conflicts for a given stream.
-     */
-    Set<Object> getConflictSetFor(UUID streamId) {
-        return conflicts.computeIfAbsent(streamId, x -> new HashSet<>());
+    /** Get a hash for the object, given a proxy. */
+    public static byte[] generateHashFromObject(ICorfuSMRProxyInternal p, Object o) {
+        return p.getSerializer().hash(o);
     }
 
     /** Get the hashed conflict set.
      * @return              The hashed conflict set.
      */
-    Map<UUID, Set<byte[]>> getHashedConflictSet() {
+    public Map<UUID, Set<byte[]>> getHashedConflictSet() {
         return conflicts.entrySet().stream()
                 .collect(Collectors.toMap(
                         // Key = UUID
-                        e -> e.getKey(),
-                        // Value = big endian byte array of hashCode.
-                        v -> v.getValue().stream()
-                                .map(o -> o.hashCode())
-                                .map(Utils::intToBigEndianByteArray)
-                                .collect(Collectors.toSet())));
+                        e -> e.getKey().getStreamID(),
+                        // Value = Generated hash.
+                        e -> e.getValue().stream()
+                                .map(o -> ConflictSetInfo.generateHashFromObject(e.getKey(), o))
+                                .collect(Collectors.toSet()),
+                        // Merge function, in case key was already mapped
+                        // (If two proxies have the same stream).
+                        (v1,v2) -> {
```

Comment:
They are keyed by proxy now. So it's possible that there are two proxies with the same Id, so they would create a duplicate key. This function handles that.